### PR TITLE
[WIP] TIFF+AIFF basic support

### PR DESCRIPTION
**Pull request by JeromeMartinez**
For the moment highly theoretical, but demonstrates that RAWcooked can handle different A/V formats in input.

### CODE REVIEW

File: Source/Lib/TIFF/TIFF.cpp [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+        return false;
+
+    Buffer_Offset = 0;
+    uint32_t Magic = Get_B4();
+    switch (Magic)
+    {
+        case 0x49492A00: // II*
+            IsBigEndian = false;
+            break;
+        case 0x4D4D002A: // MM *
```
Comment (retokromer) [inline Source/Lib/TIFF/TIFF.cpp]: _Stricto sensu_ it’s a NUL not a space, however for readability I would suggest to use `"MM *"`, and just above `"II* "`, with quotation marks and spaces.

[_Sensu lato_ the person who used NUL values in a magic number should be severely punished.]

Comment (JeromeMartinez) [reply Source/Lib/TIFF/TIFF.cpp]: Actually, the 2A00/002A thing is a number ("Answer to the Ultimate Question of Life, the Universe, and Everything"), expected to be a "version number" (if I understand well the origins of TIFF) despite the fact it is always the same number in any version up to now ("An arbitrary but carefully chosen number that further identifies the file as a TIFF file" in TIFF6).
Changed to `"II" + 42 in 16-bit hex` (and `"MM" + 42 in 16-bit hex`  respectively).

More seriously, by the way I added TIFF RGB/RGBA 8/16-bit support + AIFF/AIFC support (PCM only, Signed/Unsigned, LE/BE, 8/16/24-bit, 1/2/6-channel), still need to have more checks e.g. rejecting when pixel strips are not in a consecutive order (it is doable to support such file, but longer development and sponsors did not provide such file up to now).

**Comment by kieranjol** [comment]
Hi Jerome, does this PR support 16bit TIFF? I saw in the original commit
that you'd mentioned 8bit but your reply to Reto also mentioned 16 bit?

**Comment by JeromeMartinez** [comment]
> Hi Jerome, does this PR support 16bit TIFF?

It supports 16-bit (at least the files I have received) now, with my last update. Here are the "flavors" in current code:
```
Licensed TIFF flavors:
No  TIFF/Raw/RGB/8bit/SingleStrip
No  TIFF/Raw/RGB/16bit/SingleStrip/LE
No  TIFF/Raw/RGBA/8bit/SingleStrip
No  TIFF/Raw/RGBA/16bit/SingleStrip/LE
```

Note that this is "LE" (Little Endian), I have no 16-bit TIFF file in BE (Big Endian) but would be easy to add when I receive a 16-bit BE TIFF.

> that you'd mentioned 8bit 

I expected to do several PR but actually it is easier in 1 shot, so 8/16-bit now (+ AIFF for audio, so I can well test PCM in big endian).

File: Source/Lib/TIFF/TIFF.cpp (hunk 1 above)
Comment (retokromer) [reply Source/Lib/TIFF/TIFF.cpp]: >"An arbitrary but carefully chosen number that further identifies the file as a TIFF file"

He he! I presume «carefully chosen» means 4 + 2 = 6
;-)

>Changed to "II" + 42 in 16-bit hex (and "MM" + 42 in 16-bit hex respectively).

That’s much better, for code readability. Thank you!

**Comment by JeromeMartinez** [comment]
@retokromer thanks but still a lot to do, in order to reject unsupported TIFF files e.g. multiple strips, IFD offsets inside visual content offsets...

**Comment by retokromer** [comment]
@JeromeMartinez Indeed, yet this step allows to start testing TIFF 16-bit.

**Comment by JeromeMartinez** [comment]
I answer to myself:

> Note that this is "LE" (Little Endian), I have no 16-bit TIFF file in BE (Big Endian) but would be easy to add when I receive a 16-bit BE TIFF.

Found one but with planes (RRRR....RRRR, GGGG...) instead of packed bytes (RGBRGB....), I'll add support of planes as well as 16BE.

**Comment by JeromeMartinez** [comment]
Just added RGB 16-bit BE ("chunky format", I found 1 file with "chunky format" and spec says "Planar format is not currently in widespread use and it is not recommended for general interchange. It is used as an extension and Baseline TIFF readers are not required to support it.", so I don't prioritize development)

**Pull request merged**